drivers/pinctrl/pinctrl_sy1xx.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 sensry.io
+ */
+
+#define DT_DRV_COMPAT sensry_sy1xx_pinctrl
+
+#include <zephyr/drivers/pinctrl.h>
+
+#include <soc.h>
```

Comment:
soc.h never needed, it exists for CMSIS glue code

Reply:
actually I need the soc.h for defining SY1XX_PAD_CONFIG_ADDR

should I copy from soc.h, what is the best way for doing it?

Reply:
should be solved